UI/obs-app.cpp

```diff
@@ -1510,6 +1510,34 @@ static uint64_t convert_log_name(bool has_prefix, const char *name)
 	return std::stoull(timestring.str());
 }
 
+/* If upgrading from an older (non-XDG) build of OBS, move config files to XDG directory. */
+#if defined(__FreeBSD__)
+static void move_to_xdg(void)
+{
+	char old_path[512];
+	char new_path[512];
+	char *home = getenv("HOME");
+	if (!home)
+		return;
+
+	if (snprintf(old_path, sizeof(old_path), "%s/.obs-studio", home) <= 0)
+		return;
+
+	/* make base xdg path if it doesn't already exist */
+	if (GetAppConfigPath(new_path, sizeof(new_path), "") <= 0)
+		return;
+	if (os_mkdirs(new_path) == MKDIR_ERROR)
+		return;
+
+	if (GetAppConfigPath(new_path, sizeof(new_path), "obs-studio") <= 0)
+		return;
+
+	if (os_file_exists(old_path) && !os_file_exists(new_path)) {
+		rename(old_path, new_path);
```

Comment:
Usually we opt to:

* Not delete old files (just in case people want to downgrade or recover from other issues)
* And _if_ we choose to rename we explicitly copy first, then delete (and not do it in a single step)

Reply:
I restored the code that previously existed to handle this transition on Linux, and would rather leave it as is. If it's not sufficient now (and I understand if the bar is higher than in 2015) I'd rather just leave it out, and just mention the config path change in a release note.

For the second bullet, if we do keep this migration then a recursive copy-and-delete seems more error-prone than the atomic rename.

Reply:
As this affects FreeBSD exclusively, I'll leave this to your judgement.

Reply:
In that case my preference is to proceed with the pull request as is. When I created the "Remove non-XDG code" change I expected that (the few FreeBSD OBS) users would have to just move their config manually (and originally mentioned that in the commit message). A bit later I found the `move_to_xdg` function that was written years ago for exactly the same upgrade case in Linux.